corehq/tests/noseplugins/cmdline_params.py

```diff
@@ -0,0 +1,44 @@
+"""
+A plugin to accept parameters used for various test runner operations.
+"""
+from nose.plugins import Plugin
+
+REUSE_DB_HELP = """
+To be used in conjunction with the environment variable REUSE_DB=1.
+reset: Drop existing test dbs, then create and migrate new ones, but do not
+    teardown after running tests. This is convenient when the existing databases
+    are outdated and need to be rebuilt.
+flush: Flush all objects from the old test databases before running tests.
+    Much faster than `reset`.
+migrate: Migrate the test databases before running tests.
+teardown: Skip database setup; do normal teardown after running tests.
+"""
```

Comment:
@dimagi/team-commcare-hq 
Hey CommCareHQ team - this is a change to REUSE_DB.  The environment variable is now only a boolean - so `REUSE_DB=1` if you want to reuse the db and `REUSE_DB=0` (or just don't set it) if you don't.

The various options such as `reset` and `migrate` are now specified via a command-line parameter.  This new parameter is `--reusedb=migrate` and has four options as described in the code above or via `--help`.

This PR introduces a new option, called `flush`, which will delete all documents in couch and postgres (feel free to extend if you come across more it should clean).  This is useful for when you're working on a test and made a mistake in a `setUpClass` for instance, leaving behind test data that then gives you errors like "user X already exists!".  Functionally it's similar to `reset`, except it takes around 5 seconds to run, rather than the minute or more a normal db rebuild takes.  Rather than deleting the databases, rebuilding them, and running migrations, this essentially deletes all rows in the databases.

`reset` is still nice to have around as a big hammer if the test db just gets all screwed up, but my hope is that `flush` and `migrate` will replace it in day-to-day use.